pkg/cluster/manager.go

```diff
@@ -1034,13 +1118,18 @@ func (m *Manager) Deploy(
 		logDir := clusterutil.Abs(globalOptions.User, inst.LogDir())
 		// Deploy component
 		// prepare deployment server
+		deployDirs := []string{
+			deployDir, logDir,
+			filepath.Join(deployDir, "bin"),
+			filepath.Join(deployDir, "conf"),
+			filepath.Join(deployDir, "scripts"),
+		}
+		if globalOptions.TLSEnabled {
+			deployDirs = append(deployDirs, filepath.Join(deployDir, "tls"))
```

Comment:
The `tls` directory should be deleted when destroying component with retaining data, see https://github.com/pingcap/tiup/blob/9a83b5f1513917c4561c759f9e2235d55f11e80c/pkg/cluster/operation/destroy.go#L312